File: scrudbeans-api/src/main/java/com/github/manosbatsis/scrudbeans/api/mdd/model/IdentifierAdapter.java
package com.github.manosbatsis.scrudbeans.api.mdd.model;

/**
 * Provides read/write access to the singular identifier of an (entity) type
 */
public interface IdentifierAdapter<T, ID> {
    String getIdName(T resource);

    ID readId(T resource);

    void writeId(T resource, ID id);
}

File: scrudbeans-api/src/main/kotlin/com/github/manosbatsis/scrudbeans/api/mdd/model/IdentifierAdapter.kt
package com.github.manosbatsis.scrudbeans.api.mdd.model

/**
 * Provides information and helper functions
 * related to the identifier of an entity (type).
 */
interface IdentifierAdapter<T : Any, S : Any> {

    val entityType: Class<T>
    val entityIdType: Class<S>
    val isCompositeId: Boolean

    fun getId(resource: Any?): S?

    fun getIdAsString(resource: Any?): String?

    fun buildIdFromString(from: String?): S?

    fun convertIdToString(resourceId: S?): String?
}
